oc_mirror.sh

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+set -ex
+
+# Utility functions to create a local mirror registry using 'mirror-registry' and mirror
+# a release using 'oc mirror'
+
+function add_auth_to_pull_secret() {
+ 
+   quay_auths=`echo -n "${QUAY_REGISTRY_USER}:${QUAY_REGISTRY_PASS}" | base64 -w0`
+
+   tmpauthfile=$(mktemp --tmpdir "quayauth--XXXXXXXXXX")
+   _tmpfiles="$_tmpfiles $tmpauthfile"
+   tmppullsecret=$(mktemp --tmpdir "pullsecret--XXXXXXXXXX")
+   _tmpfiles="$_tmpfiles $tmppullsecret"
+
+
+   cat > "${tmpauthfile}" << EOF
+{
+  "auths": {
+    "${LOCAL_REGISTRY_DNS_NAME}:${LOCAL_REGISTRY_PORT}": {
+      "auth": "$quay_auths"
+    }
+  }
+}
+EOF
+
+   jq -s '.[0] * .[1]' ${tmpauthfile} ${PULL_SECRET_FILE} > ${tmppullsecret}
+   cp ${tmppullsecret} ${PULL_SECRET_FILE}
+
+   # Note that this must also be manually added to .docker/config.json, we don't currently do this in dev-scripts to avoid writing
+   # to user directory.
```

Comment:
Couldn't be created a backup file before modifying it (with a log message)? It's a little bit annoying the manual modification

Reply:
I'm still kind of resistant about modifying a file in the user home dir, I just added a check to error out if it wasn't defined in that file. But if people feel strongly about changing this file I can add it the same way we modify the pull secret file (which isn't in the home dir).
Note there is a RFE to pass the auth file into oc-mirror https://issues.redhat.com/browse/OCPBUGS-241, this would allow us to use a file not in the user home dir, it looks like there is resistance to relying on this file.

Reply:
OK I ended up adding the virthost auth to $HOME/.docker/config.json and adding an env setting for this file and allowing it be changed.  Hopefully this will help reduce the manual changes and confusion around this file.